src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_validators.py

```diff
@@ -50,20 +52,27 @@ def validate_registry_name_create(namespace):
 def validate_storage_account_name(namespace):
     client = storage_client_factory().storage_accounts
 
-    if namespace.storage_account_name is None:
+    if namespace.storage_account_name:
+        storage_account_name = namespace.storage_account_name
+        if arm_get_storage_account_by_name(storage_account_name) is None:
+            raise CLIError('The storage account {} does not exist'.format(storage_account_name))
+    else:
         while True:
             storage_account_name = str(uuid.uuid4()).replace('-', '')[:24]
             if client.check_name_availability(storage_account_name).name_available is True: #pylint: disable=E1101
                 namespace.storage_account_name = storage_account_name
-                break
+                logger.warning(
+                    'New storage account with name %s will be created and used.',
+                    storage_account_name)
+                return
 
 def validate_resource_group_name(namespace):
-    client = get_arm_service_client()
-
     if namespace.resource_group_name:
-        if not client.resource_groups.check_existence(namespace.resource_group_name):
-            parameters = ResourceGroup(location=namespace.location)
-            client.resource_groups.create_or_update(namespace.resource_group_name, parameters)
+        client = get_arm_service_client()
+        resource_group_name = namespace.resource_group_name
+
+        if not client.resource_groups.check_existence(resource_group_name):
```

Comment:
I wonder what other RPs do for these things. Do they let RP figure it out or check it on client side? What's the guidance here?